src/common.c

```diff
@@ -305,3 +306,43 @@ void raftNodeIdToString(char *output, const char *dbid, raft_node_id_t raft_id)
 {
     snprintf(output, RAFT_SHARDGROUP_NODEID_LEN + 1, "%.32s%08x", dbid, raft_id);
 }
+
+/*
+ * returns the deny_oom status for a particular set of commands
+ */
+bool isDenyOomStatus(RedisRaftCtx *rr, RaftRedisCommandArray *cmds)
+{
+    for (int i = 0; i < cmds->len; i++) {
+        RaftRedisCommand *cmd = cmds->commands[i];
+
+        if (cmd->argc < 1) {
+            continue;
+        }
+
+        const CommandSpec *cs = CommandSpecGet(cmd->argv[0]);
+        if (cs->flags & CMD_SPEC_DENYOOM) {
+            return true;
+        }
+
+    }
+
+    return false;
+}
+
+/* need to be able to normalize redis strings for dict lookups */
+char *toLowerString(const char *cmd_str, size_t cmd_len, char *buf)
+{
+    char *lcmd = buf;
+
+    if (cmd_len >= sizeof(buf)) {
+        lcmd = RedisModule_Alloc(cmd_len + 1);
```

Comment:
There are a few problems here. If `cmd_len` is larger than `sizeof(buf)`, it will allocate memory and we need to free later. Also, `sizeof(buf)` is just `8` as it is `char*`. Previously, it was a local buffer.

Probably, we don't need to make this a function as we don't call it anywhere else.  I suggest reverting it.

Reply:
Agree with reverting